python/tskit/trees.py

```diff
@@ -4549,6 +4549,68 @@ def subset(self, nodes, record_provenance=True):
         tables.subset(nodes, record_provenance)
         return tables.tree_sequence()
 
+    def union(
+        self,
+        other,
+        node_mapping,
+        check_shared_equality=True,
+        add_populations=True,
+        record_provenance=True,
+    ):
+        """
+        Returns an expanded tree sequence which contains the node-wise union of
+        ``self`` and ``other``, obtained by adding the non-shared portions of
+        ``other`` onto ``self``. The "shared" portions are specified using a
+        map that specifies which nodes in ``other`` are equivalent to those in
+        ``self``: the ``node_mapping`` argument should be an array of length
+        equal to the numberof nodes in ``other``, whose entries are the ID of
+        the matching node in ``self``, or ``tskit.NULL`` if there is no
+        matching node. Those nodes in ``other`` that map to ``tskit.NULL`` will
+        be added to ``self``, along with:
+
+        1. Individuals whose nodes are new to ``self``.
+        2. Edges whose parent or child are new to ``self``.
+        3. Mutations whose nodes are new to ``self``.
+        4. Sites which were not present in ``self``, if the site contains a newly
+           added mutation.
+
+        Nodes that are added onto ``self`` are consistently assigned new
+        populations. You may disable this behavior by setting
+        ``add_populations=False``.
+
+        Note that this operation also sorts the resulting tables, so the
+        resulting tree sequence may not be equal to ``self`` even if nothing
+        new was added (although it would differ only in ordering of the tables).
+
+        :param TableCollection other: Another table collection.
+        :param list node_mapping: An array of node IDs that relate nodes in
+            ``other`` to nodes in ``self``: the k-th element of ``node_mapping``
+            should be the index of the equivalent node in ``self``, or
```

Comment:
This can be shorter, since it's explained in detail above? How about end the sentence at the ":"?